### PR TITLE
Add RGB matrix support for Kyria

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

This change adds `g_led_config` struct to Kyria, allowing LEDs to be addressed by `RGB_MATRIX_ENABLE` features if users enable support in their own configuration.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by filterpaper** [comment]
RGB matrix transport is still in the develop tree, this will not work properly until that is merged into master.

**Review by drashna — APPROVED** [review]

**Comment by filterpaper** [comment]
Rebased changes to `develop` to reflect new path.

File: keyboards/splitkb/kyria/rev1/config.h [hunk 1]
```diff
@@ -72,3 +72,14 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+
+/* RGB matrix support */
+#ifdef RGB_MATRIX_ENABLE
+#    define SPLIT_TRANSPORT_MIRROR
+#    define DRIVER_LED_TOTAL RGBLED_NUM // Number of LEDs
+#    define RGB_MATRIX_SPLIT { 10, 10 }
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 170
+#    ifndef RGB_DISABLE_WHEN_USB_SUSPENDED
+#        define RGB_DISABLE_WHEN_USB_SUSPENDED true
+#    endif
```
Comment (drashna) [inline keyboards/splitkb/kyria/rev1/config.h:84]:
```suggestion
#    define RGB_DISABLE_WHEN_USB_SUSPENDED
```

File: keyboards/splitkb/kyria/rev1/rules.mk [hunk 2]
```diff
@@ -1,3 +1,5 @@
 OLED_DRIVER_ENABLE = yes   # Enables the use of OLED displays
 ENCODER_ENABLE = yes       # ENables the use of one or more encoders
 RGBLIGHT_ENABLE = yes      # Enable keyboard RGB underglow
+RGB_MATRIX_ENABLE = no     # Enable keyboard RGB matrix (do not use together with RGBLIGHT_ENABLE)
+RGB_MATRIX_DRIVER = WS2812 # RGB matrix driver support
```
Comment (drashna) [inline keyboards/splitkb/kyria/rev1/rules.mk:5]:
```suggestion
RGB_MATRIX_DRIVER = WS2812 # RGB matrix driver support

RGB_MATRIX_SUPPORTED = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**